File: python_solution/medium/Solution2816.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
from typing import Optional

from medium.ListNode import ListNode


class Solution:
    def doubleIt(self, head: Optional[ListNode]) -> Optional[ListNode]:
        head1= ListNode(0)
        head1.next = head

        left = head1
        right = head

        while right!=None:
            if  (right.val*2)>=10:
                left.val +=1
            right.val = (right.val*2)%10
            left = left.next
            right = right.next
        return head1 if head1.val>0 else head

File: Js_solution/medium/Solution2816.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var doubleIt = function(head) {
    let head1 = new ListNode(0);
    head1.next = head

    let left = head1
    let right = head
    while (right) {
        if (right.val*2>=10){
            left.val =left.val+ 1;
        }
        right.val = (right.val*2)%10;
        left=left.next;
        right=right.next;
    }
    return head1.val>0?head1:head;
};